src/toil_scripts/batch_alignment/bwa_alignment.py

```diff
@@ -460,43 +467,55 @@ def add_readgroups(job, job_vars, bam_id):
                   'SM={}'.format(uuid)]
     docker_call(tool='quay.io/ucsc_cgl/picardtools:1.95--dd5ac549b95eb3e5d166a5e310417ef13651994e',
                 tool_parameters=parameters, work_dir=work_dir, java_opts='-Xmx15G', sudo=sudo)
-    ids['out_bam'] = job.fileStore.writeGlobalFile(os.path.join(work_dir, output_file))
+
     # Either write file to local output directory or upload to S3 cloud storage
     if input_args['output_dir']:
         copy_to_output_dir(work_dir=work_dir, output_dir=input_args['output_dir'], files=[output_file])
     if input_args['s3_dir']:
-        job.addChildJobFn(upload_to_s3, job_vars, disk='80G')
+        job.addChildJobFn(upload_to_s3, input_args, output_file, disk='80G')
 
 
-def upload_to_s3(job, job_vars):
+def upload_to_s3(job, input_args, output_file):
     """
     Uploads a file to S3 via S3AM with encryption.  This pipeline assumes all
     BAMS contain patient data and must be encrypted upon upload.
 
     job_vars: tuple         Contains the dictionaries: input_args and ids
     """
-    # Unpack variables
-    input_args, ids = job_vars
-    uuid = input_args['uuid']
-    key_path = input_args['ssec']
+
     work_dir = job.fileStore.getLocalTempDir()
+
     # Parse s3_dir to get bucket and s3 path
     s3_dir = input_args['s3_dir']
     bucket_name = s3_dir.lstrip('/').split('/')[0]
     bucket_dir = '/'.join(s3_dir.lstrip('/').split('/')[1:])
-    base_url = 'https://s3-us-west-2.amazonaws.com/'
-    url = os.path.join(base_url, bucket_name, bucket_dir, uuid + '.bam')
-    # Retrieve file to be uploaded
-    job.fileStore.readGlobalFile(ids['out_bam'], os.path.join(work_dir, uuid + '.bam'))
-    # Generate keyfile for upload
-    with open(os.path.join(work_dir, uuid + '.key'), 'wb') as f_out:
-        f_out.write(generate_unique_key(key_path, url))
-    # Upload to S3 via S3AM
-    s3am_command = ['s3am',
-                    'upload',
-                    '--sse-key-file', os.path.join(work_dir, uuid + '.key'),
-                    'file://{}'.format(os.path.join(work_dir, uuid + '.bam')),
-                    os.path.join('s3://', bucket_name, bucket_dir, uuid + '.bam')]
+    base_url = 'https://s3.amazonaws.com/'
+    url = os.path.join(base_url, bucket_name, bucket_dir, output_file)
+    
+    # does this need to be uploaded with encryption?
```

Comment:
> does this need to be uploaded with encryption?

Almost always in my experience.  BAMs are generally considered protected data, so unless the raw sequencing data isn't under any protection then you could forego encryption.